src/transport.ts

```diff
@@ -178,28 +161,22 @@ export class PinoSentryTransport {
   }
 };
 
-export function createWriteStreamAsync(options?: PinoSentryOptions): PromiseLike<stream.Transform> {
+export function createWriteStream(options?: PinoSentryOptions): stream.Duplex {
   const transport = new PinoSentryTransport(options);
   const sentryTransformer = transport.transformer();
 
-  const pumpAsync = pify(pump);
-  return pumpAsync(
-    process.stdin.pipe(writeToStdout()),
+  return new pump(
     split((line) => {
       try {
         return JSON.parse(line);
       } catch (e) {
-        throw Error('logs should be in json format');
+        // Returning undefined will not run the sentryTransformer
+        return;
       }
     }),
     sentryTransformer
   );
 };
 
-
-export function createWriteStream(options?: PinoSentryOptions): stream.Transform & { transport: PinoSentryTransport } {
-  const transport = new PinoSentryTransport(options);
-  const sentryParse = transport.parse.bind(transport);
-
-  return Object.assign(split(sentryParse), { transport });
-};
+// Duplicate to not break API
+export const createWriteStreamAsync = createWriteStream;
```

Comment:
Cool. backward compatibility
I Will change it in future major stable version :)